### ISSUE TITLE
Deployment to Wildfly fails without `jetty-server` on the classpath

### ISSUE THREAD

**Issue by scottfrederick** [spring-projects/spring-boot#20876]
Deployment of a trivial Spring Boot application to Wildfly fails with the latest Boot snapshots unless `org.eclipse.jetty:jetty-server` is included as a dependency in the app. Wildfly has the following error information on deployment of the app: 

```
"{\"WFLYCTL0080: Failed services\" => {\"jboss.deployment.unit.\\\"spring-boot.war\\\".POST_MODULE\" => \"WFLYSRV0153: Failed to process phase POST_MODULE of deployment \\\"spring-boot.war\\\"
		Caused by: java.lang.RuntimeException: WFLYSRV0177: Error getting reflective information for class io.micrometer.core.instrument.binder.jetty.TimedHandler$2 with ClassLoader ModuleClassLoader for Module \\\"deployment.spring-boot.war\\\" from Service Module Loader
		Caused by: java.lang.NoClassDefFoundError: Failed to link io/micrometer/core/instrument/binder/jetty/TimedHandler (Module \\\"deployment.spring-boot.war\\\" from Service Module Loader): org/eclipse/jetty/util/component/Graceful\"}}"
``` 

The root cause seems to be a Micrometer dependency on a Jetty class - `io.micrometer.core.instrument.binder.jetty.TimedHandler implements org.eclipse.jetty.util.component.Graceful`. 

Deployment to Tomcat, TomEE, and Liberty work fine without `jetty-server` on the classpath.

**Comment by philwebb** [comment]
From the log

```
2020-04-07 23:55:28,849 WARN  [org.jboss.modules.define] (MSC service thread 1-8) Failed to define class io.micrometer.core.instrument.binder.jetty.TimedHandler in Module "deployment.spring-boot.war" from Service Module Loader: java.lang.NoClassDefFoundError: Failed to link io/micrometer/core/instrument/binder/jetty/TimedHandler (Module "deployment.spring-boot.war" from Service Module Loader): org/eclipse/jetty/util/component/Graceful
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1095)
	at org.jboss.modules.ModuleClassLoader.doDefineOrLoadClass(ModuleClassLoader.java:424)
	at org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:555)
	at org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:339)
	at org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:126)
	at org.jboss.modules.Module.loadModuleClass(Module.java:731)
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:247)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at java.base/java.lang.Class.getDeclaredFields0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredFields(Class.java:3061)
	at java.base/java.lang.Class.getDeclaredFields(Class.java:2248)
	at org.jboss.as.server@11.0.0.Final//org.jboss.as.server.deployment.reflect.ClassReflectionIndex.<init>(ClassReflectionIndex.java:72)
	at org.jboss.as.server@11.0.0.Final//org.jboss.as.server.deployment.reflect.DeploymentReflectionIndex.getClassIndex(DeploymentReflectionIndex.java:70)
	at org.jboss.as.ee@19.0.0.Final//org.jboss.as.ee.metadata.MethodAnnotationAggregator.runtimeAnnotationInformation(MethodAnnotationAggregator.java:57)
	at org.jboss.as.ee@19.0.0.Final//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.handleAnnotations(InterceptorAnnotationProcessor.java:106)
	at org.jboss.as.ee@19.0.0.Final//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.processComponentConfig(InterceptorAnnotationProcessor.java:91)
	at org.jboss.as.ee@19.0.0.Final//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.deploy(InterceptorAnnotationProcessor.java:76)
	at org.jboss.as.server@11.0.0.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:176)
	at org.jboss.msc@1.4.11.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
	at org.jboss.msc@1.4.11.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
	at org.jboss.msc@1.4.11.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1982)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.base/java.lang.Thread.run(Thread.java:834)

2020-04-07 23:55:28,851 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-8) MSC000001: Failed to start service jboss.deployment.unit."spring-boot.war".POST_MODULE: org.jboss.msc.service.StartException in service jboss.deployment.unit."spring-boot.war".POST_MODULE: WFLYSRV0153: Failed to process phase POST_MODULE of deployment "spring-boot.war"
	at org.jboss.as.server@11.0.0.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:183)
	at org.jboss.msc@1.4.11.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
	at org.jboss.msc@1.4.11.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
	at org.jboss.msc@1.4.11.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1982)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.RuntimeException: WFLYSRV0177: Error getting reflective information for class io.micrometer.core.instrument.binder.jetty.TimedHandler$OnCompletionListener with ClassLoader ModuleClassLoader for Module "deployment.spring-boot.war" from Service Module Loader
	at org.jboss.as.server@11.0.0.Final//org.jboss.as.server.deployment.reflect.DeploymentReflectionIndex.getClassIndex(DeploymentReflectionIndex.java:78)
	at org.jboss.as.ee@19.0.0.Final//org.jboss.as.ee.metadata.MethodAnnotationAggregator.runtimeAnnotationInformation(MethodAnnotationAggregator.java:57)
	at org.jboss.as.ee@19.0.0.Final//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.handleAnnotations(InterceptorAnnotationProcessor.java:106)
	at org.jboss.as.ee@19.0.0.Final//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.processComponentConfig(InterceptorAnnotationProcessor.java:91)
	at org.jboss.as.ee@19.0.0.Final//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.deploy(InterceptorAnnotationProcessor.java:76)
	at org.jboss.as.server@11.0.0.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:176)
	... 8 more
Caused by: java.lang.NoClassDefFoundError: Failed to link io/micrometer/core/instrument/binder/jetty/TimedHandler (Module "deployment.spring-boot.war" from Service Module Loader): org/eclipse/jetty/util/component/Graceful
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1095)
	at org.jboss.modules.ModuleClassLoader.doDefineOrLoadClass(ModuleClassLoader.java:424)
	at org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:555)
	at org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:339)
	at org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:126)
	at org.jboss.modules.Module.loadModuleClass(Module.java:731)
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:247)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at java.base/java.lang.Class.getDeclaredFields0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredFields(Class.java:3061)
	at java.base/java.lang.Class.getDeclaredFields(Class.java:2248)
	at org.jboss.as.server@11.0.0.Final//org.jboss.as.server.deployment.reflect.ClassReflectionIndex.<init>(ClassReflectionIndex.java:72)
	at org.jboss.as.server@11.0.0.Final//org.jboss.as.server.deployment.reflect.DeploymentReflectionIndex.getClassIndex(DeploymentReflectionIndex.java:70)
	... 13 more
```

**Comment by philwebb** [comment]
This turns out to be a WildFly issue. I've raised [WFLY-13345](https://issues.redhat.com/browse/WFLY-13345) and also submitted a work-around PR to Micrometer.

**Comment by scottfrederick** [comment]
Micrometer PR is https://github.com/micrometer-metrics/micrometer/pull/1976

**Comment by scottfrederick** [comment]
This is resolved with the latest Micrometer `1.5.0-SNAPSHOT`.
